contracts/exchangeIssuance/ExchangeIssuanceLeveragedForCompound.sol

```diff
@@ -0,0 +1,1346 @@
+     * If both tokens are the same, does nothing
+     *
+     * @param _collateralToken        Address of collateral token
+     * @param _collateralTokenAmount  Amount of colalteral token to swap
+     * @param _outputToken            Address of the ERC20 token to swap into
+     * @param _minAmountOutputToken   Minimum amount of output token to return to the user
+     * @param _swapData               Data (token addresses and fee levels) describing the swap path
+     *
+     * @return Amount of output token obtained
+     */
+    function _swapCollateralForOutputToken(
+        address _collateralToken,
+        uint256 _collateralTokenAmount,
+        address _outputToken,
+        uint256 _minAmountOutputToken,
+        DEXAdapter.SwapData memory _swapData
+    )
+    internal
+    isValidPath(_swapData.path, _collateralToken, _outputToken)
+    returns (uint256)
+    {
+        return addresses.swapExactTokensForTokens(
+            _collateralTokenAmount,
+            _minAmountOutputToken,
+            _swapData
+        );
+    }
+
+    /**
+     * Deposit collateral to compound to obtain collateralCToken for issuance
+     *
+     * @param _collateralToken              Address of collateral token
+     * @param _depositAmount                Amount to deposit
+     */
+    function _depositToCompound(
+        address _collateralToken,
+        uint256 _depositAmount
+    ) internal {
+        // get cTokenAddress from underlying assets
+        address cTokenAddress = _getCTokenAddess(_collateralToken);
+
+        if (_collateralToken != addresses.weth) {
+            IERC20(_collateralToken).approve(cTokenAddress, _depositAmount);
+            ICErc20Delegator(cTokenAddress).mint(_depositAmount);
+        } else {
+            IWETH(addresses.weth).withdraw(_depositAmount);
+            ICEther(payable(cTokenAddress)).mint{value: _depositAmount}();
+            // return mintTokens;
+        }
+    }
+
+    /**
+     * Convert collateralCToken from set redemption to collateralToken by withdrawing underlying from Compound
+     *
+     * @param _collateralToken       Address of the collateralToken to withdraw from Compound
+     * @param _collateralAmount      Amount of collateralToken to withdraw
+     */
+    function _withdrawFromCompound(
+        address _collateralToken,
+        uint256 _collateralAmount
+    ) internal returns (uint256){
+        uint256 result;
+
+        // get cTokenAddress from underlying assets
+        address cTokenAddress = _getCTokenAddess(_collateralToken);
+        
+        Exp memory exchangeRate = Exp({mantissa: ICEther(payable(cTokenAddress)).exchangeRateStored()});
+        uint256 cTokenAmount = div_((_collateralAmount), exchangeRate);
+
+        uint256 balance = ICErc20Delegator(cTokenAddress).balanceOf(address(this));
+        require(balance >= cTokenAmount, "ExchangeIssuance: Collateral Amount issue");
```

Comment:
I don't think we need this check, since the withdrawal from compound would fail anyway. Even though it might be a slight help in debugging I don't think it is worth all that used gas. (external contract call + additional variable + require)

Reply:
addressed in 2bc43dfd123d9c826c22e1931048af595bb3ebc3